mltsp/celery_task_tools.py

```diff
@@ -0,0 +1,177 @@
+from sklearn.ensemble import RandomForestClassifier as RFC
+from sklearn.externals import joblib
+import os
+from mltsp import cfg
+from mltsp import custom_exceptions
+import numpy as np
+
+
+def create_and_pickle_model(data_dict, featureset_key, model_type,
+                            in_docker_container):
+    """Create scikit-learn RFC model object and save it to disk.
+
+    Parameters
+    ----------
+    data_dict : dict
+        Dictionary containing features data (key 'features') and
+        class list (key 'classes').
+    featureset_key : str
+        RethinkDB ID of associated feature set.
+    model_type : str
+        Abbreviation of the type of classifier to be created.
+    in_docker_container : bool
+        Boolean indicating whether function is being called from within
+        a Docker container.
+
+    """
+    # Build the model:
+    # Initialize
+    ntrees = 1000
+    njobs = -1
+    rf_fit = RFC(n_estimators=ntrees, max_features='auto', n_jobs=njobs)
+    print("Model initialized.")
+
+    # Fit the model to training data:
+    print("Fitting the model...")
+    rf_fit.fit(data_dict['features'], data_dict['classes'])
+    print("Done.")
+    del data_dict
+
+    # Store the model:
+    print("Pickling model...")
+    foutname = os.path.join(
+        ("/tmp" if in_docker_container else cfg.MODELS_FOLDER),
+        "%s_%s.pkl" % (featureset_key, model_type))
+    joblib.dump(rf_fit, foutname, compress=3)
+    print(foutname, "created.")
+    return foutname
+
+
+def read_data_from_csv_file(fname, sep=',', skip_lines=0):
```

Comment:
What about just something like:

    import csv
    r = csv.reader(f)
    all_rows = list(r)[skip_lines:]
    colnames = all_rows[0]
    data_rows = all_rows[1:]
    data_rows = [[el if el != '?' else '0.0'for el in row] for row in x]
    return colnames, data_rows